core/domain/state_domain.py

```diff
@@ -916,8 +940,1085 @@ def get_all_outcomes(self) -> List[Outcome]:
+                        # `IsInclusivelyBetween` rule.
+                        if rule_value_a > rule_value_b:
+                            raise utils.ValidationError(
+                                f'The rule \'{rule_spec_index}\' of answer '
+                                f'group \'{ans_group_index}\' having '
+                                f'rule type \'IsInclusivelyBetween\' '
+                                f'have `a` value greater than `b` value '
+                                f'in NumericInput interaction.'
+                            )
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value_a,
+                            rule_value_b, True, True
+                        )
+                    except Exception:
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsInclusivelyBetween\' contains string '
+                            f'value - \'{rule_value_a}\', \'{rule_value_b}\''
+                        )
+
+                for range_ele in ranges:
+                    if self._is_enclosed_by(range_var, range_ele):
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' will never be '
+                            f'matched because it is made redundant '
+                            f'by the above rules'
+                        )
+
+                ranges.append(range_var)
+
+    def _validate_fraction_input(self) -> None:
+        """Validates the following in FractionInput interaction -
+        - Rule should not be duplicate
+        - Denominator should be greater than 0
+        - Solution should be in simplest form if the `simplest form`
+        setting is turned on
+        - Solution should be in proper form if the `proper form`
+        setting is turned on
+        - `IsExactlyEqualTo` rule should not have integral value when
+        `allow non zero integers` setting is off
+        - All rules should have solutions that is not subset of
+        previous rules' solutions
+        - `HasDenominatorEqualTo` rule should not have value less than 0
+        - `HasFractionalPartExactlyEqualTo` rule should always come before
+        `HasDenominatorEqualTo` rule where the fractional denominator is
+        equal to `HasDenominatorEqualTo` rule value
+
+        Raises:
+            ValidationError. Raises a validation error when the above
+                mentioned checks fails.
+        """
+        ranges: List[InteractionInstance.RangeVariable] = []
+        matched_denominator_list: List[
+            InteractionInstance.MatchedDenominator] = []
+        rule_spec_till_now: List[state_domain.RuleSpecDict] = []
+        inputs_with_whole_nums = [
```

Comment:
shouldn't this variable be named `inputs_without_fractions`? The `HasNoFractionalPart` doesn't really fit into the `inputs_with_whole_nums` name

Reply:
Done